File: 1400-Construct K Palindrome Strings.py
from collections import Counter
class Solution(object):
    def canConstruct(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: bool
        """
        if len(s) < k:
            return False
        
        charCount = Counter(s)
        oddCount = 0
        
        for value in charCount.values():
            if value%2 != 0:
                oddCount += 1
        
        return k >= oddCount
        

File: C++/1400-Construct K Palindrome Strings.cpp
class Solution {
public:
    bool canConstruct(string s, int k) {
        int N = s.size();
        if (k > N) return false;
        vector<int> cnt(26, 0);

        for (char c:s) {
            cnt[c-'a']++;
        }

        int single = 0;
        for (int i=0; i<26; i++) {
            if (cnt[i]%2 == 1) single++;
        }
        if (single > k) return false;

        return true;
    }
};